x-pack/plugins/endpoint/server/plugin.test.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { CoreSetup } from 'kibana/server';
+import { EndpointPlugin, EndpointPluginSetupDependencies } from './plugin';
+import { coreMock } from '../../../../src/core/server/mocks';
+import { PluginSetupContract } from '../../features/server';
+
+describe('test endpoint plugin', () => {
+  let plugin: EndpointPlugin;
+  let mockCoreSetup: MockedKeys<CoreSetup>;
+  let mockedEndpointPluginSetupDependencies: jest.Mocked<EndpointPluginSetupDependencies>;
+  let mockedPluginSetupContract: jest.Mocked<PluginSetupContract>;
+  beforeEach(() => {
+    plugin = new EndpointPlugin(
+      coreMock.createPluginInitializerContext({
+        cookieName: 'sid',
+        sessionTimeout: 1500,
+      })
+    );
+
+    mockCoreSetup = coreMock.createSetup();
+    mockedPluginSetupContract = {
+      registerFeature: jest.fn(),
+      getFeatures: jest.fn(),
+      getFeaturesUICapabilities: jest.fn(),
+      registerLegacyAPI: jest.fn(),
+    };
+    mockedEndpointPluginSetupDependencies = { features: mockedPluginSetupContract };
+  });
+
+  it('test properly setup plugin', async () => {
```

Comment:
My understanding of this test is that it should check that a feature has been registered and that routes have been registered. If these behaviors were tested via API integration tests, we could test the behavior instead of checking that certain methods were called. The tests wouldn't be as coupled to the implementation details. It also wouldn't requiring writing mocks that are specific to the new platform implementation (which has been changing very frequently recently.)

How would you feel about writing an API integration test that calls one of our HTTP APIs. You could also write a test that creates a user with a certain role and then with that user checks that feature registration is working correctly by making API requests (or by asserting that elements are present in the UI.)

Reply:
> requiring writing mocks that are specific to the new platform implementation (which has been changing very frequently recently.)

This is actually a reason to write such test, to make it clear that the dependencies in the code are covered at a minimum. You want to detect when you code is affected by other code changes. I will add integration test.